File: Healthcare-app/android/MDBRealmDoctor/app/src/main/java/com/wekanmdb/storeinventory/model/practitioner/Practitioner.kt
package com.wekanmdb.storeinventory.model.practitioner

import com.wekanmdb.storeinventory.model.embededclass.Attachment
import com.wekanmdb.storeinventory.model.embededclass.Human_Name
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.bson.types.ObjectId
import java.util.*

open class Practitioner(
    @PrimaryKey var _id: ObjectId = ObjectId(),
    var about: String? = null,
    var active: Boolean? = null,
    var birthDate: Date? = null,
    var gender: String? = null,
    var identifier: String? = null,
    var name: Human_Name? = null,
    var photo: RealmList<Attachment> = RealmList()
): RealmObject() {
}

File: Healthcare-app/ios/MDBRealmDoctor/MDBRealmDoctor/Practitioner.swift
//
//  Practitioner.swift
//  MDBRealmDoctor
//
//  Created by Karthick TM on 02/05/22.
//

import Foundation
import RealmSwift

class Practitioner: Object {
    
    @Persisted(primaryKey: true) var _id: ObjectId
    @Persisted var about: String?
    @Persisted var active: Bool?
    @Persisted var birthDate: Date?
    @Persisted var gender: String?
    @Persisted var identifier: String?
    @Persisted var name: Human_Name?
    @Persisted var photo: List<Attachment>
    
    override init() {}
    
    convenience init(_id: ObjectId,about: String?,active: Bool?, birthDate: Date?, gender: String?, identifier: String?, name: Human_Name?,photo: List<Attachment>) {
        self.init()
        
        self._id        = _id
        self.about      = about
        self.active     = active
        self.birthDate  = birthDate
        self.gender     = gender
        self.identifier = identifier
        self.name       = name
        self.photo      = photo
    }
}
